cmd/lakectl/cmd/docs.go

```diff
@@ -0,0 +1,148 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+// language=markdown
+var cliReferenceHeader = `---
+layout: default
+title: Command (CLI) Reference
+description: lakeFS comes with its own native CLI client. Here you can see the complete command reference.
+parent: Reference
+nav_order: 3
+has_children: false
+---
+
+# Commands (CLI) Reference
+{: .no_toc }
+
+## Table of contents
+{: .no_toc .text-delta }
+
+1. TOC
+{:toc}
+
+## Installing the lakectl command locally
+
+` + "`" + `lakectl` + "`" + ` is distributed as a single binary, with no external dependencies - and is available for MacOS, Windows and Linux.
+
+[Download lakectl](../downloads.md){: .btn .btn-green target="_blank"}
+
+
+### Configuring credentials and API endpoint
+
+Once you've installed the lakectl command, run:
+
+` + "```" + `bash
+lakectl config
+# output:
+# Config file /home/janedoe/.lakectl.yaml will be used
+# Access key ID: AKIAIOSFODNN7EXAMPLE
+# Secret access key: ****************************************
+# Server endpoint URL: http://localhost:8000/api/v1
+` + "```" + `
+
+This will setup a ` + "`" + `$HOME/.lakectl.yaml` + "`" + ` file with the credentials and API endpoint you've supplied.
+When setting up a new installation and creating initial credentials (see [Quick start](../quickstart/index.md)), the UI
+will provide a link to download a preconfigured configuration file for you.
+
+`
+
+func printOptions(buf *bytes.Buffer, cmd *cobra.Command) error {
+	flags := cmd.NonInheritedFlags()
+	flags.SetOutput(buf)
+	if flags.HasAvailableFlags() {
+		buf.WriteString("#### Options\n\n```\n")
+		flags.PrintDefaults()
```

Comment:
Does this skip `Hidden` flags?

Reply:
// HasAvailableFlags returns a bool to indicate if the FlagSet has any flags
// that are not hidden.

Reply:
OK, cool.  So actually `PrintDefaults` [skips all hidden flags](https://github.com/spf13/pflag/blob/85dd5c8bc61cfa382fecd072378089d4e856579d/flag.go#L691).

Resolving, thanks!